### PR TITLE
Brand release/6.0-maui branch appropriately

### PR DESCRIPTION
**Pull request by directhex**
Update branch used for maui previews and get it building

### CODE REVIEW

**Review by steveisok — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

File: eng/Versions.props [hunk 1]
```diff
@@ -7,8 +7,8 @@
     <MinorVersion>0</MinorVersion>
     <PatchVersion>1</PatchVersion>
     <SdkBandVersion>6.0.100</SdkBandVersion>
-    <PreReleaseVersionLabel>servicing</PreReleaseVersionLabel>
-    <PreReleaseVersionIteration></PreReleaseVersionIteration>
+    <PreReleaseVersionLabel>mauipreview</PreReleaseVersionLabel>
+    <PreReleaseVersionIteration>1</PreReleaseVersionIteration>
```
Comment (mmitche) [inline eng/Versions.props:11]: You also want to turn off `StabilizePackageVersion` (set to false) here: https://github.com/dotnet/runtime/pull/61792/files#diff-1ea18ff65faa2ae6fed570b83747086d0317f5e4bc325064f6c14319a9c4ff67R16

**Review by mmitche — COMMENTED** [review]
Aside from the StabilizePackageVersions issue, I **think** this is correct.

**Review by safern — COMMENTED** [review]
You also want to remove all instances of `ServicingVersion` and `GeneratePackageOnBuild` https://github.com/dotnet/runtime/blob/e9036b04357e8454439a0e6cf22186a0cb19e616/src/libraries/System.Text.Json/src/System.Text.Json.csproj#L13 those should be added until you want to service that specific package.

**Comment by safern** [comment]
> GeneratePackageOnBuild

Well the transport packages should keep that if you need to service those.

File: eng/packaging.targets [hunk 2]
```diff
@@ -29,17 +29,19 @@
     <GeneratePackageOnBuild Condition="('$(BuildAllConfigurations)' == 'true' or
                                        '$(IsRIDSpecificProject)' == 'true') and
                                        '$(PreReleaseVersionLabel)' != 'servicing' and
+                                       '$(PreReleaseVersionLabel)' != 'mauipreview' and
                                        '$(GitHubRepositoryName)' != 'runtimelab'">true</GeneratePackageOnBuild>
     <GeneratePackageOnBuild Condition="'$(BuildAllConfigurations)' != 'true' and
                                        '$(IsRIDSpecificProject)' != 'true' and
-                                       '$(PreReleaseVersionLabel)' == 'servicing' and
+                                       ('$(PreReleaseVersionLabel)' == 'servicing' or
```
[the remark this answers is not in the document]
Comment (steveisok) [reply eng/packaging.targets:36]: We release preview versions of OOB packages, right?

[the remark this answers is not in the document]
Comment (safern) [reply eng/packaging.targets:36]: Yes we release previews of OOB packages.

File: eng/Versions.props [hunk 3]
```diff
@@ -7,13 +7,13 @@
     <MinorVersion>0</MinorVersion>
     <PatchVersion>1</PatchVersion>
```
Comment (safern) [inline eng/Versions.props:8]: The `PatchVersion` needs to be 0 right?

Comment (mmitche) [reply eng/Versions.props:8]: For package precedence purposes, this should match the current patch level in the servicing releases.

Comment (safern) [reply eng/Versions.props:8]: So everytime we bump the `PatchVersion` on `release/6.0` we will need to bump it on this branch.

Comment (mmitche) [reply eng/Versions.props:8]: Yeah. Or just merge from release/6.0. That is also important to avoid regressing the maui branch.

Comment (directhex) [reply eng/Versions.props:8]: The number is taken from release/6.0 - should it be lower than that branch? https://github.com/dotnet/runtime/blob/release/6.0/eng/Versions.props#L8

File: eng/packaging.targets [hunk 4]
```diff
@@ -39,7 +39,7 @@
     <UseIntellisenseDocumentationFile Condition="'$(UseIntellisenseDocumentationFile)' == '' and Exists('$(LibIntellisenseDocumentationFilePath)')">true</UseIntellisenseDocumentationFile>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(PreReleaseVersionLabel)' == 'servicing'">
+  <PropertyGroup Condition="'$(PreReleaseVersionLabel)' == 'servicing' or '$(PreReleaseVersionLabel)' == 'mauipreview'">
     <!-- If no servicing version is set we need to default to 0 in order for dependency versions to
     be calculated propertly, if we don't set it to 0, we would get the dependency version using the
     product Patch Version -->
```
Comment (safern) [inline eng/packaging.targets:45]: You don't really need this for the branch as we should be using the "default" assembly version we have that uses `Major.Minor.0.Patch`.

File: eng/Versions.props (hunk 3 above)
Comment (safern) [reply eng/Versions.props:8]: > Yeah. Or just merge from release/6.0. That is also important to avoid regressing the maui branch.

Makes sense.

**Comment by directhex** [comment]
Failing because `_PackageVersion is` empty but I don't see why - it should be populated by `$(PackageVersion)` in https://github.com/directhex/runtime/blob/brand-maui-branch/src/libraries/workloads-testing.targets#L5 as far as I can see, and PackageVersion is not empty as I read the binlogs

**Comment by steveisok** [comment]
> https://github.com/directhex/runtime/blob/brand-maui-branch/src/libraries/workloads-testing.targets#L5

@radical can you provide any insight?

**Comment by directhex** [comment]
Seems more likely related to changes by @mmitche to the version handling in that file

**Comment by safern** [comment]
I see what's going on and I hit this on main when porting these changes to main. `PackageVersion` is not set at this point when this property group is evaluated, so that it is why its value is empty. In order to fix this, we need to move the `PropertyGroup` that sets `_PackageVersion` inside the target, so that the property value is assigned when the target is executed and therefore `PackageVersion` will already be evaluated and assigned. See: https://github.com/dotnet/runtime/blob/main/eng/testing/workloads-testing.targets#L96-L97

**Comment by directhex** [comment]
@safern @mmitche any objections? We're green

**Review by safern — APPROVED** [review]

**Pull request merged**